bson/bsoncodec/struct_tag_parser.go

```diff
@@ -117,3 +121,19 @@ var DefaultStructTagParser StructTagParserFunc = func(sf reflect.StructField) (S
 
 	return st, nil
 }
+
+// JSONFallbackStructTagParser has the same behavior as DefaultStructTagParser
+// but will also fallback to parsing the json tag instead on a field where the
+// bson tag isn't available.
+var JSONFallbackStructTagParser StructTagParserFunc = func(sf reflect.StructField) (StructTags, error) {
+	key := strings.ToLower(sf.Name)
+	tag, ok := sf.Tag.Lookup("bson")
+	if !ok {
```

Comment:
Ah, at first I thought this could ignore an error if  `Lookup("bson")` failed, but the docs indicate that `ok` is only false if the tag is not present. I suppose `Tag.Lookup` does not error.

Is this compatible with mgo's fallback behavior? E.g. both `bson` and `json` tags, e.g. ` bson:"name" json:"omitempty"` could potentially be a sensible combination.

Reply:
yup, this is the same as mgo's behavior. They also use bson tags and then can optionally fallback to json tags, but they don't use a combo.

Reply:
@kevinAlbs As an FYI, returning a boolean `ok` value is a common Go idiom for map lookups. The native map type supports this: `val, ok := map[key]` so other APIs that do lookups usually mimic it.